packages/web-worker/src/endpoint.ts

```
@@ -101,12 +109,36 @@ export function createEndpoint<T>(
     activeApi.get(apiCall[API_ENDPOINT]),
   );
 
+  messageEndpoint.addEventListener('message', function listener({data}) {
+    if (data == null) {
+      return;
+    }
+
+    if (TERMINATE in data) {
+      if (messageEndpoint.terminate) {
+        terminated = true;
+
+        // This should terminate the worker and continue but it hangs and tests fails waiting on the completion of this
+        // In the browser it seems like this silently fails, no errors or relevant logs.
+        // Subsequent worker calls don't throw errors either, they just silently fail
+
+        // messageEndpoint.terminate();
```

why this hang and silently fail :/